### PR TITLE
NativeShell enhancements for Jellyfin Media Player

### PR DESCRIPTION
**Pull request by iwalton3**
This allows the stock jellyfin-web to work as a web client in Jellyfin Media Player.

**Changes**
 - Add player property `useServerPlaybackInfoForAudio` that forces server-side media decisions for music.
 - Add player property `useFullSubtitleUrls` that resolves relative external subtitles URLs into full URLs for MPV.
 - Add player property `syncPlayWrapAs` which tells SyncPlay how to wrap plugins in NativeShell.
 - Add support to SyncPlay for `currentTimeAsync` method required to get accurate time from MPV.
 - Update the mouse movement checkers to also use `clientX`/`clientY` as QtWebEngine returns zeros otherwise.
 - Make PlaybackPermissionManager honor the autoplay capabilities reported by NativeShell.
 - Allow partial implementation of NativeShell interface.

### CODE REVIEW

File: src/components/apphost.js [hunk 1]
```diff
@@ -275,7 +275,7 @@ const supportedFeatures = function () {
      */
 function doExit() {
     try {
-        if (window.NativeShell) {
+        if (window.NativeShell && window.NativeShell.AppHost.exit) {
```
Comment (thornbill) [inline src/components/apphost.js:278]: I would suggest using optional chaining for this type of checks.

```suggestion
        if (window.NativeShell?.AppHost?.exit) {
```

File: src/components/syncPlay/ui/playbackPermissionManager.js [hunk 2]
```diff
@@ -32,8 +34,12 @@ class PlaybackPermissionManager {
      * Tests playback permission. Grabs the permission when called inside a click event (or any other valid user interaction).
      * @returns {Promise} Promise that resolves succesfully if playback permission is allowed.
      */
-    check () {
-        return new Promise((resolve, reject) => {
+    async check () {
```
Comment (thornbill) [inline src/components/syncPlay/ui/playbackPermissionManager.js:37]: async/await doesn't seem necessary here. I think you could just return `Promise.resolve(true)` below and remove the await when returning the promise.

File: src/components/upnextdialog/upnextdialog.js [hunk 3]
```diff
@@ -159,17 +159,21 @@ import '../../assets/css/flexstyles.scss';
...
-        void elem.offsetWidth;
-
-        elem.classList.add('upNextDialog-hidden');
-
         const fn = onHideAnimationComplete.bind(instance);
         instance._onHideAnimationComplete = fn;
 
-        dom.addEventListener(elem, transitionEndEventName, fn, {
-            once: true
+        const e = await new Promise((resolve) => {
```
Comment (thornbill) [inline src/components/upnextdialog/upnextdialog.js:165]: A more descriptive variable name might be good here instead of "e".

**Review by thornbill — CHANGES_REQUESTED** [review]
Looks good! Just a couple minor notes.

**Review by thornbill — APPROVED** [review]
:+1: lgtm. Have you tested this with any of the other wrapper apps?

**Comment by iwalton3** [comment]
I have tested with JMP and the web app only.

**Pull request merged**